### PR TITLE
Management cluster plugin: Add support for extra manifests

### PR DESCRIPTION
**Pull request by HanFa**
### What this PR does / why we need it

Required by and cherry-picked from the Oracle Support, https://github.com/vmware-tanzu/tanzu-framework/pull/3557, documented in https://github.com/vmware-tanzu/tanzu-framework/pull/3600

This PR allows extra directory of manifests to be applied at bootstrap. To be used as an escape hatch for custom TKrs.

Co-authored-by: Fang Han <fhan@vmware.com>
Signed-off-by: Naadir Jeewa <jeewan@vmware.com>

Example Usage: 
```
tanzu management-cluster create -f ~/mc.yaml --additional-tkg-system-manifests ~/tkr-oci
```

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note
Add support for applying extra manifests during management cluster bootstrap. A directory of manifests can be applied to the tkg-system namespace just before cluster creation using `--additional-tkg-system-manifests`
```

### CODE REVIEW

File: tkg/clusterclient/clusterclient.go [hunk 1]
```diff
@@ -1412,24 +1430,38 @@ func (c *client) kubectlApplyFile(url string) error {
 	if c.currentContext != "" {
 		args = append(args, "--context", c.currentContext)
 	}
-
-	args = append(args, "-f", url)
+	if o.recursive {
+		args = append(args, "--recursive")
+	}
+	if o.namespace != "" {
```
Comment (XudongLiuHarold) [inline tkg/clusterclient/clusterclient.go:1436]: Why need to specify the namespace?

**Review by XudongLiuHarold — APPROVED** [review]
Generally LGTM

**Comment by DanielXiao** [comment]
There are still failed checks.

File: tkg/clusterclient/clusterclient.go (hunk 1 above)
Comment (randomvariable) [reply tkg/clusterclient/clusterclient.go:1436]: The TKR manifests don't actually include the namespace but they need to be installed to tkg-system.

**Comment by HanFa** [comment]
> There are still failed checks.

looks like it is flake from CAPD pipeline. it succeeds after retry

**Review by DanielXiao — APPROVED** [review]

**Pull request merged**